rudder-agent/SOURCES/check-rudder-agent

```diff
@@ -166,88 +163,114 @@ check_and_fix_cf_lock() {
 
 check_and_fix_rudder_uuid() {
 
-# Default variable about UUID backup
-LATEST_BACKUPED_UUID=""
-
-# Generate a UUID if we don't have one yet
-if [ ! -f ${UUID_FILE} ]; then
-  if [ -d ${BACKUP_DIR} ]; then
-    LATEST_BACKUPED_UUID=`ls -v1 ${BACKUP_DIR}uuid-*.hive | tail -n1`
-  fi
-  if [ "z${LATEST_BACKUPED_UUID}" != "z" ]; then
-    echo_n "WARNING: The UUID of the node does not exist. The lastest backup (${LATEST_BACKUPED_UUID}) will be recovered..."
-    ${CP_A} ${LATEST_BACKUPED_UUID} ${UUID_FILE} >/dev/null 2>&1
-    echo " Done"
+  # Default variable about UUID backup
+  LATEST_BACKUPED_UUID=""
+  
+  # Generate a UUID if we don't have one yet
+  if [ ! -f "${UUID_FILE}" ]; then
+    if [ -d "${BACKUP_DIR}" ]; then
+      LATEST_BACKUPED_UUID=`ls -v1 ${BACKUP_DIR}uuid-*.hive | tail -n1`
+    fi
+    if [ "${LATEST_BACKUPED_UUID}" != "" ]; then
+      echo_n "WARNING: The UUID of the node does not exist. The lastest backup (${LATEST_BACKUPED_UUID}) will be recovered..."
+      ${CP_A} ${LATEST_BACKUPED_UUID} ${UUID_FILE} >/dev/null 2>&1
+      echo " Done"
+    else
+      echo_n "WARNING: The UUID of the node does not exist and no backup exist. A new one will be generated..."
+      uuidgen > ${UUID_FILE}
+      echo " Done"
+    fi
   else
-    echo_n "WARNING: The UUID of the node does not exist and no backup exist. A new one will be generated..."
-    uuidgen > ${UUID_FILE}
-    echo " Done"
-  fi
-else
-  # UUID is valid only if it has been generetaed by uuidgen or if it is set to 'root' for policy server
-  REGEX=`x="[a-f0-9][a-f0-9][a-f0-9][a-f0-9]" && echo "$x$x-$x-$x-$x-$x$x$x"`
-  CHECK_UUID=`cat /opt/rudder/etc/uuid.hive | egrep "^$REGEX|^root" | wc -l`
-  # If the UUID is not valid, regenerate it
-  if [ ${CHECK_UUID} -ne 1 ]; then
-    echo_n "INFO: Creating a new UUID for Rudder as the existing one is invalid..."
-    # Keep a backup of UUID even if it is not valid
-    mkdir -p /var/backups/rudder
-    cp -f /opt/rudder/etc/uuid.hive /var/backups/rudder/uuid-`date +%Y%m%d`.hive
-    # Generate a new one
-    uuidgen > ${UUID_FILE}
-    echo " Done."
+    # UUID is valid only if it has been generetaed by uuidgen or if it is set to 'root' for policy server
+    REGEX=`x="[a-f0-9][a-f0-9][a-f0-9][a-f0-9]" && echo "$x$x-$x-$x-$x-$x$x$x"`
+    CHECK_UUID=`cat /opt/rudder/etc/uuid.hive | egrep "^$REGEX|^root" | wc -l`
+    # If the UUID is not valid, regenerate it
+    if [ ${CHECK_UUID} -ne 1 ]; then
+      echo_n "INFO: Creating a new UUID for Rudder as the existing one is invalid..."
+      # Keep a backup of UUID even if it is not valid
+      mkdir -p "${BACKUP_DIR}"
+      cp -f /opt/rudder/etc/uuid.hive ${BACKUP_DIR}/uuid-`date +%Y%m%d`.hive
+      # Generate a new one
+      uuidgen > ${UUID_FILE}
```

Comment:
It was there before.
In fact the PATH has been modified at the beginning of the script to include the version  embedded in Rudder